### ISSUE TITLE
"Check failed: IsAligned()" for pluggable devices with custom device memory allocator 

### ISSUE THREAD

**Issue by kulinseth** [tensorflow/tensorflow#49202]
**Describe the current behavior**
We are using custom memory allocator registered for pluggable device. And while running unit tests like :while_v2_test.py" we are running into issue of "Check failed: IsAligned()" 
https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/framework/tensor.cc;l=678

**Describe the expected behavior**
Since we are using our own memory allocator for device allocations the alignment of the data pointer is not necessarily related to the alignment of the data on the device, making this check not applicable. What can we do to fix it so that pluggable devices with custom allocators don't run into this issue. 

We were thinking of propogating the use_bfc_allocator flag to Tensor class to bypass the alignment check with custom allocator devices. Please let us know if it makes sense. 

@mihaimaruseac , @penpornk 

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):

No this is from running the unit test case.

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
MacOS

- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
Source
- TensorFlow version (use command below):
- Python version:
3.8.2
- Bazel version (if compiling from source):
3.7.2

**Comment by yuyuan20** [comment]
Hi! I have the same issue here. Is there a fix available?

**Comment by amrka** [comment]
Try to update to tensorflow-macos==2.8.0. It seems to resolve the issue.

**Comment by AaronJny** [comment]
> Try to update to tensorflow-macos==2.8.0. It seems to resolve the issue.

Work for me! 

When I had this problem before, the version of tensorflow-macos was only updated to 2.7, and the official version of tensorflow has been updated to 2.8. 

Maybe this is fixed in tensorflow-macos 2.8?

**Comment by mihaimaruseac** [comment]
Closing as it should be resolved with TF 2.8

**Comment by nigelparsad** [comment]
Error still persists in tensorflow-macos=2.9.2 and tensorflow-metal=0.5.0.

Error occurred while attempting to run inference using a Tensorflow 2 version of the Mask R-CNN deep learning neural network architecture for instance segmentation:

2022-08-16 04:06:15.497243: F tensorflow/core/framework/tensor.cc:728] Check failed: IsAligned() ptr = 0x2d5a5cf60

Machine is an M1 Max MacBook Pro.

Message has been posted on Apple Dev Forum:  https://developer.apple.com/forums/thread/696907
